include/States.h

```diff
@@ -0,0 +1,78 @@
+#ifndef ALE_STATES_H
+#define ALE_STATES_H
+
+#include<vector>
+#include <gsl/gsl_interp.h>
+ 
+namespace ale {
+
+enum interpolation {
+  /// Interpolate using linear interpolation
+  linear,
+  /// Interpolate using spline interpolation
+  spline
+};
+
+/** A 3D cartesian vector */
+struct Vec3d {
+  double x;
+  double y;
+  double z;
+
+  Vec3d(double x, double y, double z) : x(x), y(y), z(z) {};
+  Vec3d() : x(0.0), y(0.0), z(0.0) {};
+};
+
+/** A state vector with position and velocity*/
+struct State {
+  Vec3d position;
+  Vec3d velocity;
+
+  State(ale::Vec3d position, ale::Vec3d velocity) : position(position), velocity(velocity) {}; 
+  State() {};
+};
+
+class States {
+  public:
+    // Constructors
```

Comment:
Questions for all:
Where and how are we going to document classes? class.h or class.cpp or ....?
What format are the comments in?

Reply:
I second this question. I wasn't sure what to do, so I ended up using a confusing mixture of different formats, which is certainly not ideal.

Reply:
@scsides I don't seem to be able to respond to your above comment about the algorithm in-line, but your understanding as described in your comment matches mine.

Reply:
I prefer documenting in the header files because that is what I look at if I'm developing against something.

Reply:
Agreed, IMHO, all API docs go in the header. If I want to use a class, it is nice to not have all the code cluttering up what I'm looking for. All I should need to code against an API is the libraries and headers. If I'm looking for a bug, fine, I need the source, and good internal "WHY THIS IS..." comments.
FYI, this is opposite to the current ISIS style doc.